src/sentry_plugins/trello/plugin.py

```diff
@@ -0,0 +1,278 @@
+from sentry_plugins.base import CorePluginMixin
+from .client import TrelloApiClient
+
+
+SETUP_URL = "https://github.com/getsentry/sentry/blob/master/src/sentry_plugins/trello/Trello_Instructions.md"  # NOQA
+
+
+class TrelloPlugin(CorePluginMixin, IssuePlugin2):
+    description = "Create Trello cards on issues"
+    slug = "trello"
+    title = "Trello"
+    conf_title = title
+    conf_key = "trello"
+    auth_provider = None
+    resource_links = [("Trello Setup Instructions", SETUP_URL)] + CorePluginMixin.resource_links
+
+    def get_config(self, project, **kwargs):
+        """
+        Return the configuration of the plugin.
+        Pull the value out of our the arguments to this function or from the DB
+        """
+
+        def get_value(field):
+            initial_values = kwargs.get("initial") or {}
+            if initial_values.get(field):
+                return initial_values[field]
+            return self.get_option(field, project)
+
+        token_config = {
+            "name": "token",
+            "type": "secret",
+            "label": "Trello API Token",
+            "default": None,
+        }
+
+        token_val = get_value("token")
+        if token_val:
+            token_config["required"] = False
+            token_config["prefix"] = token_val[:5]
+            token_config["has_saved_value"] = True
+        else:
+            token_config["required"] = True
+
+        key_val = get_value("key")
+
+        key_config = {
+            "name": "key",
+            "type": "text",
+            "required": True,
+            "label": "Trello API Key",
+            "default": key_val,
+        }
+
+        config = [key_config, token_config]
+        org_value = get_value("organization")
+        include_org = kwargs.get("add_additial_fields") or org_value
+        if key_val and token_val and include_org:
+            trello_client = TrelloApiClient(key_val, token_val)
+            try:
+                org_options = trello_client.get_organization_options()
+                config.append(
+                    {
+                        "name": "organization",
+                        "label": "Trello Organization",
+                        "choices": org_options,
+                        "type": "select",
+                        "required": False,
+                        "default": org_value,
+                    }
+                )
+            except Exception as e:
+                self.raise_error(e)
+        return config
+
+    def validate_config(self, project, config, actor=None):
+        """
+        Make sure the configuration is valid by trying to query for the organizations with the auth
+        """
+        trello_client = TrelloApiClient(config["key"], config["token"])
+        try:
+            trello_client.get_organization_options()
+        except Exception as e:
+            raise PluginError(self.message_from_error(e))
```

Comment:
Why not `self.raise_error(e)` here? (comment?)

Reply:
@BYK I can add a comment

Reply:
Decided to just use `self.raise_error(e)` here